### PR TITLE
Refactor shared float parsing helper

### PR DESCRIPTION
**Pull request by eXPerience83**
## Summary

Extracts a shared `parse_float` helper into `helpers.py` to eliminate duplicated float-parsing logic across `climate.py` and `sensor.py`.

### What changed
- `helpers.py`: added `parse_float(value, *, precision=None) -> float | None` — handles `None`, comma-decimal (`"23,5"`), dot-decimal (`"23.5"`), int/float values, and optional rounding via `precision`.
- `climate.py`: replaced 6 calls to `_parse_float()` with `parse_float()`; removed the static method.
- `sensor.py`: replaced `_parse_float1()` → `parse_float(precision=1)` and `_parse_float6()` → `parse_float(precision=6)`; removed both static methods.
- `tests/test_helpers.py`: added 10 pytest cases (8 parametrized + 2 independent) covering `None`, dot/comma decimals, invalid strings, int/float, precision=1, precision=6, `TypeError` handling, and unparseable collections.
- `AGENTS.md`: documented intentional Python 3.14 / PEP 758 syntax and `pytest-homeassistant-custom-component` policy.
- `requirements_test.txt`: relaxed test dep to `pytest-homeassistant-custom-component>=0.13.340`.

### Behavior
No functional changes. Climate temperatures remain unrounded; sensor temperatures stay 1-decimal; coordinates stay 6-decimal. `TypeError` and `ValueError` from upstream values continue to return `None`.

### Validations
- `ruff check --fix --select I .` — passed
- `ruff check .` — passed
- `black --check .` — 25 files left unchanged
- `pytest -q` — 96 passed
- `python -m compileall -q custom_components tests scripts` — passed

### Not changed
- No version bump
- No changelog edits
- No backend/API, device registry, unique_id, migrations, options, README, workflows changes

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved numeric value handling so temperatures and coordinates are parsed more reliably, including decimal-comma formats and rounded precision where needed.
  * Updated request handling in the integration tests to better reflect how network calls are captured.

* **Tests**
  * Added coverage for numeric parsing edge cases, invalid inputs, and rounding behavior.

* **Documentation**
  * Clarified project guidance and versioning notes for contributors.

* **Chores**
  * Updated test dependencies.

### CODE REVIEW

File: custom_components/airzoneclouddaikin/helpers.py [hunk 1]
```diff
@@ -359,6 +359,27 @@ async def async_auto_exit_sleep_if_needed(
...
+    normalized to ``"23.5"`` before parsing.  When *precision* is *None*
+    (default) the parsed float is returned without rounding; otherwise
+    ``round(result, precision)`` is applied.
+    """
+
+    if value is None:
+        return None
+    try:
+        parsed = float(str(value).replace(",", "."))
+    except TypeError, ValueError:
```
[the remark this answers is not in the document]
Comment (eXPerience83) [reply custom_components/airzoneclouddaikin/helpers.py:376]: This is a false positive. The project targets Python 3.14 only (as documented in AGENTS.md and pyproject.toml). PEP 758 allows multiple exceptions without parentheses when no `as` clause is present, e.g. `except TypeError, ValueError:`. This syntax is valid and intentional. It compiles and passes all tests under Python 3.14.5. Reviewers should not rewrite this syntax for Python 3.13 or earlier compatibility.

**Comment by eXPerience83** [comment]
Current CI status note:

The current test failures are not caused by the `parse_float()` refactor in this PR.

The failing tests are the existing HTTP API tests that still use `aioresponses`. With `pytest-homeassistant-custom-component>=0.13.340`, CI now resolves `pytest-homeassistant-custom-component 0.13.341`, which pulls Home Assistant `2026.7.0b1` and `aiohttp 3.14.1`.

`aioresponses 0.7.9` is currently incompatible with `aiohttp 3.14.x` because it builds `ClientResponse` without the new required `stream_writer` argument.

Plan:

* Keep this PR open for now.
* Wait briefly to see whether `aioresponses` publishes a fixed release for `aiohttp 3.14`.
* If `aioresponses` is updated, rerun CI with `pytest-homeassistant-custom-component>=0.13.340`.
* If no fix is released soon, we will handle it in a separate PR by either:

  * migrating the HTTP tests from `aioresponses` to `aiointercept>=0.1.7`, or
  * temporarily pinning `pytest-homeassistant-custom-component==0.13.340`.

No production integration code is affected by this dependency issue.

**Comment by eXPerience83** [comment]
@coderabbitai full review
